lib/prestashopper/api.rb

```diff
@@ -22,33 +22,65 @@ def initialize(url, key)
     # List resources that the API key can access
     # @return [Array<Symbol>] list of resources the API can access
     def resources
-      xml = @resources_res.get.body
-      xml_doc = Nokogiri::XML xml
-      nodes = xml_doc.xpath '/prestashop/api/*'
-      resources_list = []
-      nodes.each{|n| resources_list << n.name.to_sym}
-      return resources_list
-    end
-
-    # Get all products data
-    # @return [Array<Hash>] list of products. Each product is represented by a hash with all its attributes.
-    def get_products
-      # /api/products returns XML with the IDs of each individual product
-      xml_products = @resources_res['products'].get.body
-      xml_products_doc = Nokogiri::XML xml_products
-      products_nodes = xml_products_doc.xpath '/prestashop/products/*/@id'
-      ids_list = []
-      products_nodes.each{|n| ids_list << n.value}
-
-      # GET each individual product to get the whole data
-      products = []
-      ids_list.each do |id|
-        xml_product = @resources_res["products/#{id}"].get.body
-        product = Product.xml2hash xml_product
-        products << product
+      @resources ||= Nokogiri::XML(@resources_res.get.body).xpath('/prestashop/api/*').collect { |resource| resource.name.to_sym }
+    end
+
+    def method_missing(method, *args, &block)
```

Comment:
Isn't it a bit inconsistent that ```get_orders(1, 2, 3)``` returns an array of objects, each with all the data about an order, while ```get_orders( )``` returns an array of IDs? 

I think getting data about e.g. all products from prestashop without knowing their IDs beforehand is an important use case that is not covered here.